### ISSUE TITLE
CO2 conversion per MJ coal

### ISSUE THREAD

**Issue by grdw** [quintel/etsource#1203]
@ChaelKruip and I did some digging around for the issue that's currently present [here](https://github.com/quintel/atlas/pull/64#issuecomment-270400190). 

We found out that the `weighted_carrier_co2_per_mj` attribute is 0.096 here (`energy_distribution_coal`) at my local etengine:

**localhost/national_dataset/energy_distribution_coal (0.093)**
![screen shot 2017-01-04 at 17 46 15](https://cloud.githubusercontent.com/assets/2676542/21650621/d57bf3aa-d2a5-11e6-8cce-0bd1befb7810.png)

**localhost/local_dataset/energy_distribution_coal (0.096)**
![screen shot 2017-01-04 at 17 29 08](https://cloud.githubusercontent.com/assets/2676542/21649834/57fe6a04-d2a3-11e6-8195-47e894d23f89.png)

However, when you go over to etengine beta, it all of a sudden is 0.093:

**beta/national_dataset/energy_distribution_coal (0.093)**
![screen shot 2017-01-04 at 17 30 26](https://cloud.githubusercontent.com/assets/2676542/21649897/84b274dc-d2a3-11e6-8561-edb68ef70562.png)

When you do a `git log -p carriers/coal.ad` inside of etsource you find out that the commit responsible for this change is this one: 117230dd8737b4bf2401e0f16962243b450816fe 

However this commit is from August 2015 which seems a bit odd.. 

So could this be a caching problem? 

Including @jorisberkhout @antw

**Comment by antw** [comment]
> localhost/**national_dataset**/energy_distribution_coal (**0.093**)
>
> However, when you go over to etengine beta, it all of a sudden is 0.093:
> beta/**national_dataset**/energy_distribution_coal (**0.093**)

It seems to me that both your local copy and beta are in agreement about the CO<sub>2</sub> per MJ for the national dataset, which makes a caching problem unlikely (a cached value lingering from 17 months ago is also quite improbable [but not impossible] 😆 ).

The value shown on that page is rounded to two decimal places and the real value – which can be found either by hovering the mouse over it, or [on the carrier page](https://beta-engine.energytransitionmodel.com/data/621369/carriers/coal) – is 0.0926771694.

The NL region [has fuel-chain emissions data for coal](https://github.com/quintel/etsource/blob/master/datasets/nl/fce/coal.yml). The `co2_conversion_per_mj` attribute is a fallback for regions which do *not* have this data. For regions that do, this data is used to construct the `co2_converson_per_mj` of the carrier and the default value is ignored.

* **With FCE disabled in ETModel:** the `co2_converson_per_mj` FCE values from ETSource are used to construct the attribute, so that the *source* of the fuel is accounted for in CO<sub>2</sub> emissions and the carrier `co2_per_mj` represents only the emissions due to conversion.

  `V(CARRIER(coal), co2_per_mj)` = 0.0926771694

* **With FCE enabled:** the other CO<sub>2</sub> emissions attributes (exploration, extraction, treatment, etc) are also added to the `co2_conversion_per_mj` value to get the `co2_per_mj`.

  `V(CARRIER(coal), co2_per_mj)` = 0.09762395013

---

It seems there is an oversight in [Atlas::Carrier#load_fce_values](https://github.com/quintel/atlas/blob/1a67953fc974177bad9eb194750d4677b4b99880/lib/atlas/carrier.rb#L36) – which never considered that FCE values would be stored in a _different_ dataset directory (understandably?) – and should probably instead look something like:

```ruby
def load_fce_values(region)
  # Use Dataset#dataset_dir instead of manually constructing the path:
  path = Atlas::Dataset.find(region).dataset_dir.join("/fce/#{ key }.yml")
  path.file? && YAML.load_file(path)
end
```

**Comment by ploh** [comment]
I did some more digging but did not find out the cause of the problem.
What I did found out:
1. The value is 0.096 for datasets `eu` and `de` as well as for a derived dataset with `base nl` and a derived dataset with `base de`. It is 0.093 for dataset `nl`. This is all on my machine.
2. 0.093 is the very outdated value for `coal`, but also the current value for `coal_gas` and `cokes` - for which this value correctly shows up in all datasets.

I still have no idea whatsoever where the difference between `nl` and `base nl` may result from. The MsgPacked graphs are identical, the `country.ad` is identical and the dataset_dir of the derived dataset refers to that of nl.
